src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -135,18 +137,18 @@ internal void ApplySuppressionsAndStoreAnalysisResult(AnalysisScope analysisScop
 
             lazyLocalDiagnostics = lazyLocalDiagnostics ?? new Dictionary<SyntaxTree, Dictionary<DiagnosticAnalyzer, ImmutableArray<Diagnostic>.Builder>>();
 
-            foreach (var diagsByTree in diagnostics.GroupBy(d => d.Location.SourceTree))
+            foreach (var diagsByTree in diagnostics.GroupBy(d => d.Location.SourceTree!))
```

Comment:
Why is this safe?

Reply:
From discussion with @mavasani, only diagnostics that are in source get into the local queues, others are treated as non-local or compilation.
Last commit adds assertions instead of this suppression.

---
In reply to: [405838845](https://github.com/dotnet/roslyn/pull/43025#discussion_r405838845) [](ancestors = 405838845)